### PR TITLE
Feature/9.2/compat

### PR DESCRIPTION
**Pull request by orthagh**

### CODE REVIEW

File: setup.php [hunk 1]
```diff
@@ -34,37 +34,37 @@
...
-   define ("PLUGIN_ORDER_NUMBER_STEP", 1 / pow(10, $CFG_GLPI["decimal_number"]));
+   define("PLUGIN_ORDER_NUMBER_STEP", 1 / pow(10, $CFG_GLPI["decimal_number"]));
 }
 
 // Autoload
 if (!defined('PCLZIP_TEMPORARY_DIR')) {
    define('PCLZIP_TEMPORARY_DIR', GLPI_DOC_DIR . '/_tmp/pclzip');
 }
-include_once GLPI_ROOT . "/plugins/order/vendor/autoload.php";
+//include_once GLPI_ROOT . "/plugins/order/vendor/autoload.php";
```
Comment (trasher) [inline setup.php]: I guess OdtPHP stuff will not work without that.

Comment (orthagh) [reply setup.php]: i guess you're right, and walid should have been tired

**Comment by orthagh** [comment]
more coming soon

**Comment by tech62** [comment]
The dropdown list are empty, when we have to choice the template used to generate the purchase order after fresh update of this plugin, any idea ?

**Comment by orthagh** [comment]
Hello @tech62, 
Could you test the following GLPI fix:
https://github.com/glpi-project/glpi/commit/dbb8c697031

I think it's your issue.

**Comment by tech62** [comment]
I've made the changes and to be sure, i've replaced the file but no effect :/ the list stay empty
I've copy the odt file in the glpi/plugins/order/templates and /glpi/order/generate/custom.php, that work previously.

**Comment by orthagh** [comment]
@tech62 i currently testing with my customer the different view of the plugin and it seems we finished to find bug on our side.
Could you test with the last version of this pr and ideally the last version of glpi 9.2/bugfixes branch (the incoming 9.2.1) ?

And if you still have the issue, could you attach a screenshot ?

Regards

**Comment by tech62** [comment]
Hi @orthagh, i've clone the repo but problem :
```
PHP Warning: include_once(/var/www/glpi/plugins/order/vendor/autoload.php): failed to open stream: No such file or directory in /var/www/glpi/plugins/order/setup.php at line 65
PHP Warning: include_once(): Failed opening '/var/www/glpi/plugins/order/vendor/autoload.php' for inclusion (include_path='.:/usr/share/php') in /var/www/glpi/plugins/order/setup.php at line 65
```
`
Fatal error: Cannot redeclare plugin_order_check_config() (previously declared in /var/www/glpi/plugins/order/setup.php:214) in /var/www/glpi/plugins/orderbak/setup.php on line 222
`

Effectively the vendor folder is missing
idea ?

**Comment by orthagh** [comment]
2 points
- you can get composer here: https://getcomposer.org/, after just composer install in the plugin folder (and in glpi if you pull also the bgufixes version)
- the Fatal error is due to the fact you have copied your old order in the plugins folder. GLPI doesn't support this, move it elsewhere.

**Comment by tech62** [comment]
After i've install composer and move my "orderbak" folder in another location, i've always this error :
`PHP Notice: Constant PCLZIP_TEMPORARY_DIR already defined in /var/www/glpi/plugins/order/setup.php at line 64`

**Comment by orthagh** [comment]
You're on the master branch of the plugin not this one.
You can retrieve the dev version for GLPI 9.2 here:
https://github.com/pluginsGLPI/order/archive/feature/9.2/compat.zip

or if you cloned the repository, just do:
git checkout -b feature/9.2/compat origin/feature/9.2/compat

**Comment by tech62** [comment]
It's OK for the errors, but the dropdownlist is empty (after i've copied the custom.php in the generate folder and my odt file in the templates folder)

**Comment by tech62** [comment]
If i just download the compat.zip and unzip it 
`PHP Warning: include_once(/var/www/glpi/plugins/order/vendor/autoload.php): failed to open stream: No such file or directory in /var/www/glpi/plugins/order/setup.php at line 67
PHP Warning: include_once(): Failed opening '/var/www/glpi/plugins/order/vendor/autoload.php' for inclusion (include_path='.:/usr/share/php') in /var/www/glpi/plugins/order/setup.php at line 67`

**Comment by orthagh** [comment]
same as above, redo the ``composer install``

**Comment by orthagh** [comment]
Oh sorry, i miss the last part of your message.
Could you post a screenshot, please ?

**Comment by tech62** [comment]
bug_order

**Comment by orthagh** [comment]
Oh ok ! I didn't understand that was you issue.

Where did you place your folders for templates/logos etc ?
Now it should copied in glpi/_plugins/order/.

Regards

**Comment by tech62** [comment]
I found the bug, i've read this comment for this [issue](https://github.com/pluginsGLPI/order/issues/130#issuecomment-339769154).
I've make  :
- An order folder in glpi/files/_plugins/ who's missing
- And logo, templates, generate subfolders in glpi/files/_plugins/orders

With that and the odt / custom.php filein each directory and that works.

**Comment by orthagh** [comment]
so let me know if you find other issues with 9.2 branch.
Thank you

**Comment by tech62** [comment]
The last bug is in my odt file generated by the plugin : some tag aren't correctly filled (that display : {entity_phone} instead of my work phone)

**Comment by orthagh** [comment]
Could you share your custom.php ?

**Comment by tech62** [comment]
It's this line :  ` $odf->setVars('entity_phone', $entity->fields["phone"],true, 'UTF-8');` breaking my odt generation

**Comment by tech62** [comment]
Fixed by change 
`$odf->setVars('entity_phone', $entity->fields["phone"],true, 'UTF-8');`
To
`$odf->setVars('entity_phone', $entity->fields["phonenumber"],true, 'UTF-8');`

**Pull request closed without merging**